src/main/scala/uk/gov/homeoffice/drt/repository/ArrivalRepository.scala

```diff
@@ -40,32 +45,32 @@ class ArrivalRepository[F[_] : Sync](val sessionPool: Resource[F, Session[F]]) e
   implicit val logger = Slf4jLogger.getLogger[F]
 
   val decoder: Decoder[ArrivalTableData] =
-    (text ~ int4 ~ text ~ text ~ text ~ text ~ int4.opt ~ timestamp ~ timestamp.opt).map {
-      case code ~ number ~ destination ~ origin ~ terminal ~ status ~ totalpassengers ~ scheduled ~ scheduled_departure  =>
+    (text ~ int4 ~ text ~ text ~ text ~ text ~ int4.opt ~ timestamp ~ timestamp.opt ~ timestamp.opt ~ timestamp.opt ~ timestamp.opt ~ timestamp.opt ~ timestamp.opt).map {
+      case code ~ number ~ destination ~ origin ~ terminal ~ status ~ totalpassengers ~ scheduled ~ estimated ~ actual ~ estimatedchox ~ actualchox ~ pcp ~ scheduled_departure =>
         ArrivalTableData(
-          code, number, destination, origin, terminal, status, totalpassengers, scheduled, scheduled_departure
+          code, number, destination, origin, terminal, status, totalpassengers, scheduled, estimated, actual, estimatedchox, actualchox, pcp, scheduled_departure
         )
     }
 
 
-  private def selectArrivalsForADate: Query[LocalDateTime ~ LocalDateTime, ArrivalTableData] =
+  private def selectArrivalsForOriginsAndADate(origins: List[String]): Query[List[String] ~ LocalDateTime ~ LocalDateTime, ArrivalTableData] =
```

Comment:
This should presumably be `selectArrivalsForOrigins`

Reply:
Query does need date as well but origin is only pass as function parameter and dates are passed through query parameter

Reply:
I just added change , as it is parameterized query it just need list size .